src/pages/outlier-detection/OutlierDetection.js

```diff
@@ -262,53 +262,53 @@ class OutlierDetection extends Page {
         this.setState({ algorithm: value })
     }
 
-    // toggleCheckbox(element) {
-    //     const api = this.context.d2.Api.getApi()
-    //     const elements = this.state.elements
-    //     for (let i = 0; i < elements.length; i++) {
-    //         const currentElement = elements[i]
-    //         if (currentElement.key === element.key) {
-    //             this.context.updateAppState({
-    //                 pageState: {
-    //                     loading: true,
-    //                 },
-    //             })
-    //             const apiCall = currentElement.marked ? api.delete : api.post
-    //             apiCall(
-    //                 apiConf.endpoints.markDataValue,
-    //                 OutlierAnalyisTable.convertElementToToggleFollowupRequest(
-    //                     currentElement
-    //                 )
-    //             )
-    //                 .then(() => {
-    //                     if (this.isPageMounted()) {
-    //                         currentElement.marked = !currentElement.marked
-    //                         elements[i] = currentElement
-
-    //                         this.context.updateAppState({
-    //                             showSnackbar: true,
-    //                             snackbarConf: {
-    //                                 type: SUCCESS,
-    //                                 message: i18n.t(
-    //                                     currentElement.marked
-    //                                         ? 'Marked'
-    //                                         : 'Unmarked'
-    //                                 ),
-    //                             },
-    //                             pageState: {
-    //                                 elements,
-    //                                 loading: false,
-    //                             },
-    //                         })
-    //                     }
-    //                 })
-    //                 .catch(error => {
-    //                     this.manageError(error)
-    //                 })
-    //             break
-    //         }
-    //     }
-    // }
+    toggleCheckbox(element) {
+        const api = this.context.d2.Api.getApi()
+        const elements = this.state.elements
+        for (let i = 0; i < elements.length; i++) {
+            const currentElement = elements[i]
+            if (currentElement.key === element.key) {
+                this.context.updateAppState({
+                    pageState: {
+                        loading: true,
+                        elements,
+                        showTable: elements && elements.length > 0,
```

Comment:
`elements` is constant here, and `showTable` will ALWAYS be `true` since this is within a for-loop of the `elements` array - is updating these `pageState` variables needed?  Is it because the states aren't merged automatically?

Reply:
> Is it because the states aren't merged automatically

Correct:
https://github.com/dhis2/data-quality-app/blob/196ae1c17e553af523cc42e1cd49f458bb5cc154/src/App.js#L54-L65

I've made this explicit in 8353aee